### PR TITLE
Sourcery Starbot ⭐ refactored vrdmr/Computational-Investing-pt-1

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>master</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/Computational-Investing-pt-1 master
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/dialect.py [hunk 1]
```diff
@@ -154,12 +154,12 @@ def translate(url, dialectName="chef"):
 
 def test(url):
 	"""test all dialects against URL"""
+	import webbrowser
 	for dialect in ("chef", "fudd", "olde"):
 		outfile = "%s.html" % dialect
 		fsock = open(outfile, "wb")
 		fsock.write(translate(url, dialect))
 		fsock.close()
-		import webbrowser
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/dialect.py:162]: Function `test` refactored with the following changes:
- Hoist statements out of for/while loops (`hoist-statement-from-loop`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/kgp/kgp.py [hunk 3]
```diff
@@ -204,9 +204,8 @@ def do_p(self, node):
         completely ignored)
         """
         keys = node.attributes.keys()
-        if "class" in keys:
-            if node.attributes["class"].value == "sentence":
-                self.capitalizeNextWord = 1
+        if "class" in keys and node.attributes["class"].value == "sentence":
+            self.capitalizeNextWord = 1
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/kgp/kgp.py:208]: Function `KantGenerator.do_p` refactored with the following changes:
- Merge nested if conditions (`merge-nested-ifs`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage1/soundex1d.py [hunk 4]
```diff
@@ -49,13 +49,13 @@ def soundex(source):
...
     for c in source:
-        if not ('A' <= c <= 'Z') and not ('a' <= c <= 'z'):
+        if not ('A' <= c <= 'Z' or 'a' <= c <= 'z'):
             return "0000"
 
     # Soundex algorithm:
     # 1. make first character uppercase
     source = source[0].upper() + source[1:]
-    
+
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage1/soundex1d.py:58]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage1/soundex1e.py [hunk 5]
```diff
@@ -46,13 +46,13 @@ def soundex(source):
...
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
         return "0000"
 
     # Soundex algorithm:
     # 1. make first character uppercase
     source = source[0].upper() + source[1:]
-    
+
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage1/soundex1e.py:55]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage2/soundex2a.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage2/soundex2a.py:49]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage2/soundex2b.py [hunk 7]
```diff
@@ -46,7 +46,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage2/soundex2b.py:49]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage2/soundex2c.py [hunk 8]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage2/soundex2c.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3a.py [hunk 9]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3a.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3b.py [hunk 10]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3b.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3c.py [hunk 11]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3c.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3d.py [hunk 12]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage3/soundex3d.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4a.py [hunk 13]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4a.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4b.py [hunk 14]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4b.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4c.py [hunk 15]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4c.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4d.py [hunk 16]
```diff
@@ -22,7 +22,7 @@ def soundex(source):
     # Soundex requirements:
     # source string must be at least 1 character
     # and must consist entirely of letters
-    if (not source) or (not source.isalpha()):
+    if not (source and source.isalpha()):
```
Comment (SourceryAI) [inline CompInvesting1/Books+HelperCode/diveintopython-5.4/py/soundex/stage4/soundex4d.py:25]: Function `soundex` refactored with the following changes:
- Simplify logical expression using De Morgan identities (`de-morgan`)

File: CompInvesting1/HWs/HW3/marketsim.py [hunk 17]
```diff
@@ -31,62 +31,57 @@ def readOrdersFile(filename):
...
 	# The Matrix created by NumPy is Float, converting into ints
 	integerMatrix = ordersMatrix.astype(int)
-	
+
 	# Sorting the matrix by date. Sorting the first 3 columns - 
 	# First column is Year, Second- month, Third- Day
 	# Sample - 2011,1,14,AAPL,Buy,1500,
 	sortedMatrix = np.array(sorted(integerMatrix, key=tuple))
-	
+
```
Comment (SourceryAI) [inline CompInvesting1/HWs/HW3/marketsim.py:84]: Function `readOrdersFile` refactored with the following changes:
- Hoist repeated code outside conditional statement (`hoist-statement-from-if`)
- Swap if/else to remove empty if body (`remove-pass-body`)

File: CompInvesting1/HWs/HW4/marketsim.py [hunk 18]
```diff
@@ -31,62 +31,57 @@ def readOrdersFile(filename):
...
 	# The Matrix created by NumPy is Float, converting into ints
 	integerMatrix = ordersMatrix.astype(int)
-	
+
 	# Sorting the matrix by date. Sorting the first 3 columns - 
 	# First column is Year, Second- month, Third- Day
 	# Sample - 2011,1,14,AAPL,Buy,1500,
 	sortedMatrix = np.array(sorted(integerMatrix, key=tuple))
-	
+
```
Comment (SourceryAI) [inline CompInvesting1/HWs/HW4/marketsim.py:84]: Function `readOrdersFile` refactored with the following changes:
- Hoist repeated code outside conditional statement (`hoist-statement-from-if`)
- Swap if/else to remove empty if body (`remove-pass-body`)

File: CompInvesting1/HWs/HW6/bollingerEvents.py [hunk 19]
```diff
@@ -33,53 +33,35 @@ def fetchNYSEDataForEvents(dt_start, dt_end, spxversion, marketSym):
...
 	# Keys to be read from the data, it is good to read everything in one go.
 	ls_keys = ['open', 'high', 'low', 'close', 'volume', 'actual_close']
-	
+
 	# Reading the data, now d_data is a dictionary with the keys above.
 	# Timestamps and symbols are the ones that were specified before.
 	ldf_data = c_dataobj.get_data(ldt_timestamps, ls_symbols, ls_keys)
 	d_data = dict(zip(ls_keys, ldf_data))
-	
+
```
Comment (SourceryAI) [inline CompInvesting1/HWs/HW6/bollingerEvents.py:50]: Function `fetchNYSEDataForEvents` refactored with the following changes:
- Remove unreachable code (`remove-unreachable-code`)

File: CompInvesting1/HWs/HW7/marketsim.py [hunk 20]
```diff
@@ -31,62 +31,57 @@ def readOrdersFile(filename):
...
 	# The Matrix created by NumPy is Float, converting into ints
 	integerMatrix = ordersMatrix.astype(int)
-	
+
 	# Sorting the matrix by date. Sorting the first 3 columns - 
 	# First column is Year, Second- month, Third- Day
 	# Sample - 2011,1,14,AAPL,Buy,1500,
 	sortedMatrix = np.array(sorted(integerMatrix, key=tuple))
-	
+
```
Comment (SourceryAI) [inline CompInvesting1/HWs/HW7/marketsim.py:84]: Function `readOrdersFile` refactored with the following changes:
- Hoist repeated code outside conditional statement (`hoist-statement-from-if`)
- Swap if/else to remove empty if body (`remove-pass-body`)

**Review by vrdmr — APPROVED** [review]

**Pull request merged**